scripts/test/Muon/load_file_widget/loadfile_presenter_single_file_test.py

```diff
@@ -0,0 +1,299 @@
+        self.presenter.enable_multiple_files(False)
+
+        patcher = mock.patch('Muon.GUI.Common.load_file_widget.model.load_utils')
+        self.addCleanup(patcher.stop)
+        self.load_utils_patcher = patcher.start()
+
+    def mock_browse_button_to_return_files(self, files):
+        self.view.show_file_browser_and_return_selection = mock.Mock(return_value=files)
+
+    def mock_user_input_text(self, text):
+        self.view.get_file_edit_text = mock.Mock(return_value=text)
+
+    def mock_model_to_load_workspaces(self, workspaces, runs, filenames):
+        self.load_utils_patcher.load_workspace_from_filename = mock.Mock(side_effect=zip(workspaces, runs, filenames))
+
+    def load_workspaces_into_model_and_view_from_browse(self, workspaces, runs, files):
+        self.mock_model_to_load_workspaces(workspaces, runs, files)
+        self.mock_browse_button_to_return_files(files)
+
+        self.presenter.on_browse_button_clicked()
+        self.wait_for_thread(self.presenter._load_thread)
+
+    def load_failure(self):
+        raise ValueError("Error text")
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # TESTS
+    # ------------------------------------------------------------------------------------------------------------------
+
+    @run_test_with_and_without_threading
+    def test_browser_dialog_opens_when_browse_button_clicked(self):
+        self.mock_browse_button_to_return_files(["file.nxs"])
+
+        self.presenter.on_browse_button_clicked()
+        self.wait_for_thread(self.presenter._load_thread)
+
+        self.assertEqual(self.view.show_file_browser_and_return_selection.call_count, 1)
+
+    @run_test_with_and_without_threading
+    def test_loading_not_initiated_if_no_file_selected_from_browser(self):
+        self.mock_model_to_load_workspaces([], [], [])
+        self.mock_browse_button_to_return_files([])
+
+        self.presenter.on_browse_button_clicked()
+        self.wait_for_thread(self.presenter._load_thread)
+
+        self.assertEqual(self.load_utils_patcher.load_workspace_from_filename.call_count, 0)
+
+    @run_test_with_and_without_threading
+    def test_buttons_disabled_while_load_thread_running(self):
+        self.mock_browse_button_to_return_files(["file.nxs"])
+
+        self.presenter.on_browse_button_clicked()
+        self.wait_for_thread(self.presenter._load_thread)
+
+        self.assertEqual(self.view.disable_load_buttons.call_count, 1)
+        self.assertEqual(self.view.enable_load_buttons.call_count, 1)
+
+    @run_test_with_and_without_threading
+    def test_buttons_enabled_after_load_even_if_load_thread_throws(self):
+        self.model.execute = mock.Mock(side_effect=self.load_failure)
+
+        self.presenter.on_browse_button_clicked()
+        self.wait_for_thread(self.presenter._load_thread)
+
+        self.assertEqual(self.view.disable_load_buttons.call_count,
```

Comment:
check it has loaded something and that the disable/enable were called at least once